online-shop/src/app/shopping-cart/shopping-cart.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit } from '@angular/core';
+import { Order } from '../data/order';
+import { AuthService } from '../services/auth.service';
+import { ProductService } from '../services/product.service';
+
+@Component({
+  selector: 'app-shopping-cart',
+  templateUrl: './shopping-cart.component.html',
+  styleUrls: ['./shopping-cart.component.scss'],
+})
+export class ShoppingCartComponent implements OnInit {
+  orders: Order[] = [];
+  canCheckout = this.authService.userHasRole('customer');
```

Comment:
**suggestion:** you should have this logic in the component lifecycle, more specifically in `ngOnInit`